app/eth2wrap/genwrap/genwrap.go

```diff
@@ -49,13 +50,28 @@ import (
 {{- end}}
 )
 
+// Interface assertions
+var (
+    _ eth2client.Service = (*Service)(nil)
+
+	{{range .Providers}} _ eth2client.{{.}} = (*Service)(nil)
+    {{end -}}
+)
+
+type eth2Provider interface {
```

Comment:
had to introduce this interface since we cannot rely on implementation anymore since we use either eth2http or eth2multi.